ext/AdvancedPSLibtaskExt.jl

```diff
@@ -86,17 +97,18 @@ end
 # PG requires keeping all randomness for the reference particle
 # Create new task and copy randomness
 function AdvancedPS.forkr(trace::LibtaskTrace)
+    taped_globals = trace.model.ctask.taped_globals
```

Comment:
@mhauru can we make a Libtask API for this line to avoid depending on `Libtask.TapedTask` internals. One possibility is to define: 

```julia
Libtask.get_taped_globals!(T::Any, task::TapedTask)
```

which returns `taped_globals` for a given `task`.

Reply:
I think that's a good idea. The arguments are going to be a bit awkward: The one-argument version is already reserved for getting the global from task local storage with `Libtask.get_taped_globals!(T::Any)`, but the `T` argument in `Libtask.get_taped_globals!(T::Any, task::TapedTask)` is actually unnecessary.

Do you mind if I leave fixing this for a bit later? For now I've put in some helper functions in this extension to smooth this out a bit.

Reply:
Sounds good to have helper functions for this PR.